File: C++/W3Schools/6_Variables/6_Variables.cpp

// C++: W3Schools - Main Tutorial
// Section 6: Variables

#include <iostream>
using namespace std;

int main() {
   
  // CREATING VARIABLES
  int var;                                // Declaration of the variable (allocates memory).
  var = 10;                               // Assignment of the variable (give it a value).
  cout << var;                            // Prints the variable. NOTE: the name of a variable is called the identifier.

  // VARIABLE TYPES
  int myNum = 5;                          // Integer (whole number without decimals).
  double myFloatNum = 5.99;               // Floating point number (with decimals).
  char myLetter = 'D';                    // Character (a single letter).
  string myText = "Hello";                // String (text, an array of characters).
  bool myBoolean = true;                  // Boolean (true or false).

  // MULTIPLE VARIABLES
  int a = 5, b = 6, c = 50;               // We can declare many variables at once.
  cout << a + b + c;

  int x, y, z;                            // We can also assign one value to multiple variables.
  x = y = z = 50;
  cout << x + y + z;

  // CONSTANTS
  const float PI = 3.14;                  // Constants are read-only variables whose value cannot be reassigned.
  const int minutesPerHour = 60;          // As a consequence, constants must be assigned in the same line that they are declared in.

  return 0;
}

File: W3Schools/Backend/Java/Main_Tutorial/6_Variables.java

// Java: W3Schools - Main Tutorial
// Section 6: Variables

// Java Variable Types:
//      - String
//      - int
//      - float
//      - boolean
//      - char

public class Main {
  public static void main(String[] args) {

    int myNum1;    // Declare a variable
    myNum1 = 15;   // Assign a value

    // Print the value of the variable:
    System.out.println(myNum1);

    // Final variables cannot be changed once they are assigned a value:
    final int myNum2 = 15;
    
    // Example of all variable types:
    int myNum = 5;
    float myFloatNum = 5.99f;
    char myLetter = 'D';
    boolean myBool = true;
    String myText = "Hello";

    // We can also declare multiple variables in one line:
    int x1 = 5, y1 = 6, z1 = 50;
    System.out.println(x1 + y1 + z1);

    // Or declare multiple variables with the same value:
    int x2, y2, z2;
    x2 = y2 = z2 = 50;
    System.out.println(x2 + y2 + z2);

  }
}
